### ISSUE TITLE
Degenerate polygons when switching layers

### ISSUE THREAD

**Issue by ibesora** [visgl/deck.gl#5365]
#### Description
I'm getting degenerate polygons randomly when changing between two MVT layers
This is the expected result which is correctly rendered most of the time
![good](https://user-images.githubusercontent.com/2301378/105398978-44a1d780-5c23-11eb-9daa-4d4da3a4dd5d.png)
This happens randomly when going back to the same layer:
![lines](https://user-images.githubusercontent.com/2301378/105399046-5e431f00-5c23-11eb-932b-71814f7825d2.png)
![lines2](https://user-images.githubusercontent.com/2301378/105399053-60a57900-5c23-11eb-8343-380945d548ef.png)

#### Expected Behavior
To not see degenerate polygons

#### Environment
 - Framework Version: deck.gl 8.2.8
 - Browser Version: Chrome 88.0.4324.96, Latest Safari
 - OS: Ubuntu 20.04.1, OSX Big Sur

#### Logs
Nothing printed in the console

**Comment by Pessimistress** [comment]
Please provide steps to reproduce this issue.

**Comment by ibesora** [comment]
I have created a [codesandbox project](https://codesandbox.io/s/deckgl-layer-toggle-5nst1) that changes between a scatterplotlayer and an MVTLayer. This is not exactly our setup because we are also fetching the points with the MVTLayer. Unfortunately that data is private so I can't share it.
As I said before, those degenerate polygons not appear always so it's quite random but I have seen the same thing on that sandbox I'm linking.
I've tried enabling both deck.gl and luma debug modes but there's no warning or error message when that happens.
Another thing that I noticed is that if instead of using the MVTLayer to render the polygons I use the geojson layer, that doesn't seem to happen.

**Comment by ibesora** [comment]
I've just recorded a video about it to see if it gives you more clues on what's happening.
Notice that when hovering one of the polygons with a degenerated edge, the degenerate edge disappears.
Another thing that happens and that you can see at around the 23 seconds mark is that although all the big degenerated polygon gets shaded when I select the base polygon, it seems it doesn't get picked.
And finally, at around the 32 seconds mark you can see that I'm able to select a polygon by picking on a degenerated edge.

https://user-images.githubusercontent.com/2301378/105613521-58843f80-5dc3-11eb-95eb-2462c35feabf.mp4

**Comment by Pessimistress** [comment]
The CodeSandbox fails to load the MVT tiles (CORS issue).

Can you try the latest release (production 8.3.14, beta 8.4.0-beta.1)?

May be related to #5340

**Comment by ibesora** [comment]
I have updated my version to 8.3.14 and the problem is still there.
Regarding this issue being related to #5340 I have to say that leaving the app open for hours doesn't exhibit that same behaviour, it's only after redrawing the polygon layer some times that it appears. Even then it's not really consistent, redrawing the layer once those bad triangles appear might exhibit some other defects or render perfectly fine.

In regards to the [codesandbox project](https://codesandbox.io/s/deckgl-layer-toggle-5nst1) there's something weird happening with our CORS configuration. I've opened an issue with our infra team but meanwhile, if you want to try it, I've found a workaround:
1. Run the sandbox as is. You'll see that you get some fetch errors.
2. Comment lines 55 to 59 in `map.js`
3. Run again

**Comment by Pessimistress** [comment]
I can load the tiles now but have not been able to reproduce the issue. What GPU do you have?

**Comment by ibesora** [comment]
I have changed some things in our code to fix a deck.gl assertion failed error.
Reading your comment [here](https://github.com/visgl/deck.gl/issues/4176#issuecomment-576788952) I decided to recreate the layer each time instead of reusing the one I created before. Doing this makes the problem disappear so I think something might be wrong with the layer state management. 

I'll close this. Sorry for wasting your time
